src/chainstate/stacks/db/transactions.rs

```diff
@@ -2123,6 +2169,56 @@ pub mod test {
         conn.commit_block();
     }
 
+    #[test]
+    fn process_smart_contract_user_aborts_2257() {
+        let contract = "(asserts! false (err 1))";
```

Comment:
I think it would be super useful to see a test with all the different behavior - success, abort, panic - as a direct call,  in a nested `contract-call` - testing for writes being rollbacked and transaction inclusion.

Reply:
I would agree -- and think we can add more test coverage for that after the next xenon reset. If that testing turns up an issue here, I think we can figure it out at that point.